schema/iguana.owl

```diff
@@ -162,42 +195,152 @@
     <rdfs:domain rdf:resource="&iont;Suite"/>
     <rdfs:range rdf:resource="&iont;Task"/>
   </owl:ObjectProperty>
-  
+
 
   <owl:ObjectProperty rdf:about="&iprop;workerResult">
     <rdfs:label>workerResult</rdfs:label>
 	<rdfs:comment xml:lang="en">Assigns a Worker to an Task. (mostly a Stresstest)</rdfs:comment>
     <rdfs:domain rdf:resource="&iont;Task"/>
     <rdfs:range rdf:resource="&iont;Worker"/>
-  </owl:ObjectProperty>  
+  </owl:ObjectProperty>
 
 
   <owl:ObjectProperty rdf:about="&iprop;metric">
     <rdfs:label>metric</rdfs:label>
 	<rdfs:comment xml:lang="en">Annotates a Task, Worker or ExecutedQuery with a Metric. The Metric itself is provided using the Property, this just annotates the subject to provide these results.</rdfs:comment>
-    <rdfs:domain rdf:resource="&iont;Worker"/>
-    <rdfs:domain rdf:resource="&iont;Task"/>
-    <rdfs:domain rdf:resource="&iont;ExecutedQuery"/>
+    <rdfs:domain>
+      <rdfs:Class>
+        <owl:unionOf rdf:parseType="Collection">
+          <owl:Class rdf:about="&iont;Task"/>
+          <owl:Class rdf:about="&iont;Worker"/>
+          <owl:Class rdf:about="&iont;ExecutedQuery"/>
+        </owl:unionOf>
+      </rdfs:Class>
+    </rdfs:domain>
     <rdfs:range rdf:resource="&iont;Metric"/>
-  </owl:ObjectProperty>  
-  
+  </owl:ObjectProperty>
+
   <owl:ObjectProperty rdf:about="&iprop;query">
     <rdfs:label>query</rdfs:label>
 	<rdfs:comment xml:lang="en">Assigns an ExecutedQuery to a Worker or Task. The ExecutedQuery provides further metrics and statistics.</rdfs:comment>
-    <rdfs:domain rdf:resource="&iont;Worker"/>
-    <rdfs:domain rdf:resource="&iont;Task"/>
+    <rdfs:domain>
+      <owl:Class>
+        <owl:unionOf rdf:parseType="Collection">
+          <owl:Class rdf:about="&iont;Worker"/>
+          <owl:Class rdf:about="&iont;Task"/>
+        </owl:unionOf>
+      </owl:Class>
+    </rdfs:domain>
     <rdfs:range rdf:resource="&iont;ExecutedQuery"/>
-  </owl:ObjectProperty>  
-  
+  </owl:ObjectProperty>
+
   <owl:ObjectProperty rdf:about="&iprop;queryID">
     <rdfs:label>queryID</rdfs:label>
 	<rdfs:comment xml:lang="en">Assigns a Query and its statistics, as well as the string of the query to an ExecutedQuery.</rdfs:comment>
-    <rdfs:domain rdf:resource="&iont;ExecutedQuery"/>
+    <rdfs:domain>
+      <owl:Class>
+        <owl:unionOf rdf:parseType="Collection">
+          <owl:Class rdf:about="&iont;ExecutedQuery"/>
+          <owl:Class rdf:about="&iont;QueryExecution"/>
+        </owl:unionOf>
+      </owl:Class>
+    </rdfs:domain>
     <rdfs:range rdf:resource="&iont;Query"/>
-  </owl:ObjectProperty>  
+  </owl:ObjectProperty>
 
     <!-- DatatypeProperty definitions -->
 
+  <owl:DatatypeProperty rdf:about="&iprop;time">
```

Comment:
Please check for all properties if they have domain and range set. (only for vocabulary defined here not for imported)

Reply:
They should all have a domain and range. Protege doesn't show the domain and range of a property, if it has xsd:dayTimeDuration as its range for some reason.

Reply:
Change from xsd:dayTimeDuration to xsd:duration. Protege still doesn't show them, but they should be there.